pkgs/servers/sql/postgresql/buildPostgresqlExtension.nix

```diff
@@ -60,70 +60,88 @@
   lib,
   stdenv,
   postgresql,
+  nix-update-script,
 }:
 
 args:
 
 let
-  buildPostgresqlExtension = finalAttrs: prevAttrs: {
-    buildInputs = [ postgresql ] ++ prevAttrs.buildInputs or [ ];
+  buildPostgresqlExtension =
+    finalAttrs:
+    {
+      enableUpdateScript ? true,
+      ...
+    }@prevAttrs:
+    {
+      passthru =
+        prevAttrs.passthru or { }
+        // lib.optionalAttrs enableUpdateScript {
+          updateScript =
+            prevAttrs.passthru.updateScript or (nix-update-script (
+              lib.optionalAttrs (lib.hasInfix "unstable" prevAttrs.version) {
+                extraArgs = [ "--version=branch" ];
+              }
+            ));
+        };
 
-    installFlags = [
-      "DESTDIR=${placeholder "out"}"
-    ] ++ prevAttrs.installFlags or [ ];
+      buildInputs = [ postgresql ] ++ prevAttrs.buildInputs or [ ];
 
-    postInstall =
-      ''
-        # DESTDIR + pg_config install the files into
-        # /nix/store/<extension>/nix/store/<postgresql>/...
-        # We'll now remove the /nix/store/<postgresql> part:
-        if [[ -d "$out${postgresql}" ]]; then
-            cp -alt "$out" "$out${postgresql}"/*
-            rm -r "$out${postgresql}"
-        fi
+      installFlags = [
+        "DESTDIR=${placeholder "out"}"
+      ] ++ prevAttrs.installFlags or [ ];
```

Comment:
@NixOS/nix-formatting so I don't know about the current state, but after reading this diff, I'm about to get a bad feeling about our formatting efforts: the file was formatted before already, but now has another indent which makes it very hard to distinguish between indents from the formatter and actual changes.

To quote from RFC166:

> We want a unified Nix code style that's consistent with itself, easily readable, accessible and __results in readable diffs.__

(emphasis mine)

I don't think this is given at all here.

I know very well that the entire topic is pretty complex and I appreciate your efforts. I'm sorry for being rather direct here, but seeing this diff and having to expect more of those in the future makes me uncomfortable.

Reply:
Is this the same issue as https://github.com/NixOS/nixfmt/issues/228? If so, we can continue the discussion there so it's more central and not split over many PRs.